cadence/src/sinks/udp.rs

```diff
@@ -118,17 +118,29 @@ impl MetricSink for UdpMetricSink {
 pub(crate) struct UdpWriteAdapter {
     addr: SocketAddr,
     socket: UdpSocket,
+    stats: WriterStats,
```

Comment:
For consistency, we should add stats to the non-buffered `UdpMetricSink` and `UnixMetricSink` as well.